awx/ui_next/src/screens/Credential/CredentialEdit/CredentialEdit.jsx

```diff
@@ -12,45 +16,100 @@ function CredentialEdit({ credential, me }) {
+          },
+        ] = await Promise.all([
+          CredentialTypesAPI.read({
+            or__namespace: ['gce', 'scm', 'ssh'],
+          }),
+          CredentialsAPI.readInputSources(credential.id, { page_size: 200 }),
+        ]);
         setCredentialTypes(loadedCredentialTypes);
+        const inputSourcesMap = {};
+        loadedInputSources.forEach(inputSource => {
+          inputSourcesMap[inputSource.input_field_name] = inputSource;
+        });
+        setInputSources(inputSourcesMap);
       } catch (err) {
         setError(err);
       } finally {
         setIsLoading(false);
       }
     };
     loadData();
-  }, []);
+  }, [credential.id]);
 
   const handleCancel = () => {
     const url = `/credentials/${credential.id}/details`;
 
     history.push(`${url}`);
   };
 
+  const createAndUpdateInputSources = pluginInputs =>
+    Object.entries(pluginInputs).map(([fieldName, fieldValue]) => {
+      if (!inputSources[fieldName]) {
+        return CredentialInputSourcesAPI.create({
+          input_field_name: fieldName,
+          metadata: fieldValue.inputs,
+          source_credential: fieldValue.credential.id,
+          target_credential: credential.id,
+        });
+      }
+      if (fieldValue.touched) {
+        return CredentialInputSourcesAPI.update(inputSources[fieldName].id, {
+          metadata: fieldValue.inputs,
+          source_credential: fieldValue.credential.id,
+        });
+      }
+
+      return null;
+    });
+
+  const destroyInputSources = inputs => {
+    const destroyRequests = [];
+    Object.values(inputSources).forEach(inputSource => {
+      const { id, input_field_name } = inputSource;
+      if (!inputs[input_field_name]?.credential) {
+        destroyRequests.push(CredentialInputSourcesAPI.destroy(id));
+      }
+    });
+    return destroyRequests;
+  };
+
   const handleSubmit = async values => {
-    const { organization, ...remainingValues } = values;
+    const { inputs, organization, ...remainingValues } = values;
+    const pluginInputs = {};
+    Object.entries(inputs).forEach(([key, value]) => {
+      if (value.credential && value.inputs) {
+        pluginInputs[key] = value;
+        delete inputs[key];
+      }
+    });
     setFormSubmitError(null);
     try {
-      const {
-        data: { id: credentialId },
-      } = await CredentialsAPI.update(credential.id, {
-        user: (me && me.id) || null,
-        organization: (organization && organization.id) || null,
-        ...remainingValues,
-      });
-      const url = `/credentials/${credentialId}/details`;
+      await Promise.all([
+        CredentialsAPI.update(credential.id, {
+          user: (me && me.id) || null,
+          organization: (organization && organization.id) || null,
+          inputs: inputs || {},
+          ...remainingValues,
+        }),
+        ...destroyInputSources(pluginInputs),
+      ]);
+      await Promise.all(createAndUpdateInputSources(pluginInputs));
```

Comment:
i think this all looks good?  we don't have to put this in one of our useRequest-like hooks to fix the "calling stuff on unmounted component" issues do we?